packages/ui/Marquee/MarqueeDecorator.js

```diff
@@ -310,12 +309,7 @@ const MarqueeDecorator = hoc(defaultConfig, (config, Wrapped) => {
 			this.timerState = TimerState.CLEAR;
 			this.distance = null;
 			this.contentFits = false;
-		}
-
-		componentWillMount () {
-			if (this.context.Subscriber) {
-				this.context.Subscriber.subscribe('resize', this.handleResize);
-			}
+			this.registry = null;
```

Comment:
I think we need to add an `unregister` in `componentWillUnmount`. Otherwise, if an instance is removed from the tree and a resize occurs, we'll get some errors/complaints. Let's add a test for that scenario as well.